### PR TITLE
Enable negative rounding for FormattableNumbers

### PR DESCRIPTION
**Pull request by sisou**

### CODE REVIEW

File: src/formattable-number/FormattableNumber.ts [hunk 1]
```diff
@@ -72,16 +72,20 @@ export class FormattableNumber {
 
     public round(decimals: number): this {
         if (this._digits.length - this._decimalSeparatorPosition <= decimals) return this;
-        const digitsToKeep = this._digits.substring(0, this._decimalSeparatorPosition + decimals);
-        if (Number.parseInt(this._digits[this._decimalSeparatorPosition + decimals], 10) < 5) {
+        const firstRoundedIndex = this._decimalSeparatorPosition + decimals;
```
Comment (danimoh) [inline src/formattable-number/FormattableNumber.ts]: The name `firstRoundedIndex` is a bit misleading.
Before the leading 0 is added, it is the first index that gets cut off, after adding the leading 0, it's the first index that gets rounded.

File: tests/FormattableNumber.spec.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/* global describe, it, expect */
+
+import { FormattableNumber } from '../src/formattable-number/FormattableNumber';
+
+describe('FormattableNumber', () => {
+    fit('Can round numbers', () => {
```
Comment (danimoh) [inline tests/FormattableNumber.spec.ts]: Why `fit` instead of `it`?

File: tests/FormattableNumber.spec.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        expect(new FormattableNumber(12345.12355).round(2).toString()).toBe('12345.12');
+        expect(new FormattableNumber(12345.125).round(2).toString()).toBe('12345.13');
+        expect(new FormattableNumber(12345.123).round(0).toString()).toBe('12345');
+        expect(new FormattableNumber(12344.123).round(-1).toString()).toBe('12340');
+        expect(new FormattableNumber(12345.123).round(-2).toString()).toBe('12300');
+        expect(new FormattableNumber(12345.123).round(-1).toString()).toBe('12350');
+        expect(new FormattableNumber(12355.123).round(-2).toString()).toBe('12400');
```
Comment (danimoh) [inline tests/FormattableNumber.spec.ts]: Would be good to also add a test for the overflow:
```ts
expect(new FormattableNumber(99.99).round(1).toString()).toBe('100');
expect(new FormattableNumber(99.99).round(-1).toString()).toBe('100');
```
Also adding tests for rounding at the extremes would be interesting:
```ts
expect(new FormattableNumber(12.34).round(2).toString()).toBe('12.34');
expect(new FormattableNumber(12.34).round(3).toString()).toBe('12.34'); 
expect(new FormattableNumber(12.34).round(-2).toString()).toBe('0');
expect(new FormattableNumber(12.34).round(-3).toString()).toBe('0');
expect(new FormattableNumber(99).round(-2).toString()).toBe('100');
expect(new FormattableNumber(99).round(-3).toString()).toBe('0');
```

File: tests/FormattableNumber.spec.ts (hunk 2 above)
Comment (sisou) [reply tests/FormattableNumber.spec.ts]: Forgot to change it back. (`fit` stands for "focused it()).

Comment (danimoh) [reply tests/FormattableNumber.spec.ts]: That's why I was wondering why you're using it ;)
